src/sentry/integrations/utils/code_mapping.py

```diff
@@ -505,24 +523,27 @@ def find_roots(stack_path: str, source_path: str) -> tuple[str, str]:
     stack_path_delim = SLASH if SLASH in stack_path else BACKSLASH
     if stack_path_delim == BACKSLASH:
         stack_path = stack_path.replace(BACKSLASH, SLASH)
+    if (straight_path_idx := get_straight_path_prefix_end_index(stack_path)) > 0:
+        stack_root += stack_path[:straight_path_idx]
+        stack_path = stack_path[straight_path_idx:]
+
+    overlap_to_check: list[str] = stack_path.split(SLASH)
+    stack_root_items: list[str] = []
```

Comment:
In [the comment](https://github.com/getsentry/sentry/pull/67982/files/d7128e085ee6228aab8943e6a5d1aab7d00c9b00#r1544984784) of a previous PR, I explained that we changed the find_roots to iterate characters to stop an error where certain Node.js stack paths starting with `app:///` would have the stack_root be `app://` (missing a slash). This turned out to introduce another issue where an overlap match would occur too early. 

This change switches to filepath item based iteration again, and takes care of the `app:///` issue separately on loine 526-528